### PR TITLE
Use external mulesoft oauth2 client

### PR DESCRIPTION
**Pull request by zapo**
This moves api-console to use mulesoft external Oauth2.0 client "client-oauth2" and standardize the related auth strategy to expose promises like other auth strategies.

First time I work with js ecosystem (grunt & all) so please let me know if anything is off or obviously wrong.

Thanks !

### CODE REVIEW

File: dist/scripts/api-console.js [hunk 1]
```diff
@@ -1364,10 +1364,12 @@
               //// TODO: Make a uniform interface
               if (scheme && scheme.type === 'OAuth 2.0') {
                 authStrategy = new RAML.Client.AuthStrategies.Oauth2(scheme, $scope.credentials);
-                authStrategy.authenticate(request.toOptions(), function (jqXhr, err) {
-                  $scope.requestOptions = request.toOptions();
-                  handleResponse(jqXhr, err);
-                });
+                $scope.requestOptions = request.toOptions();
+
+                authStrategy.authenticate(request.toOptions()).then(
```
Comment (dmartinezg) [inline dist/scripts/api-console.js]: do not call `toOptions()` twice

File: dist/scripts/api-console.js [hunk 2]
```diff
@@ -2479,7 +2481,17 @@
     return window.open(location, 'Authentication', 'toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width=' + w + ', height=' + h + ', top=' + top + ', left=' + left);
   }
 
-  Oauth2.prototype.authenticate = function(options, done) {
+  Oauth2.prototype.popsicleParametrize = function(options) {
+    return {
+      url: options.url,
```
Comment (dmartinezg) [inline dist/scripts/api-console.js]: I know that not everybody likes it, but can you vertically align these?

**Comment by jcenturion** [comment]
@zapo: ping :)

**Comment by javok002** [comment]
Closing, because it seems to be no response.

**Pull request closed without merging**